### ISSUE TITLE
Broadcom (Brcm47xx) unbootable on newer builds >1.10

### ISSUE THREAD

**Issue by stopspazzing** [ericpaulbishop/gargoyle#899]
@lantis1008 mentioned brcm47xx builds have issues after 1.10 builds. Believe that is the last build on Chaos Calmer Openwrt base, but quote me if I am wrong.

I am using an old, decommissioned [e2000](https://openwrt.org/toh/linksys/e2000) and have serial connection to it so can debrick it if needed on testing builds.

@lantis1008 [needed serial logs so here they are](https://gist.github.com/stopspazzing/55bcfceeff6567a6fd67387a42d65ab2)

Notes: started with **1.10-generic** at the top then flashed to **1.12.0-generic** using GUI, where you see it just 100% bricked waiting for tftp, I issue some nvram show and nvram info commands for extra debugging, I then do a nvram erase, and reboot and flash **1.13.x-generic** using latest pulled files. This is where it says the irq_poll issue:
```
[   22.349075] CPU: 0 PID: 249 Comm: kmodloader Not tainted 4.14.202 #0
[   22.355544] Stack : 80490000 8044fe00 00000000 00000000 8042a4c4 81809db4 818bd95c 80477947
[   22.364120]         804265b8 000000f9 804c3ae0 11008401 804c3f40 11008400 81809d68 5a440891
[   22.372690]         00000000 00000000 804c0000 0000e2d0 00000000 000003b3 00000007 00000000
[   22.381260]         00000000 00000da2 00055393 00000000 80000000 00000000 00000005 00000005
[   22.389830]         804814c8 11008401 804c3f40 8047ce78 00000003 801f7f68 00000000 804c0000
[   22.398400]         ...
[   22.400915] Call Trace:
[   22.403453] [<8000b144>] show_stack+0x58/0x100
[   22.408025] [<80058a9c>] __report_bad_irq+0x58/0xe8
[   22.413015] [<80058e10>] note_interrupt+0x27c/0x32c
[   22.418004] [<8005631c>] handle_irq_event_percpu+0x4c/0x64
[   22.423606] [<8005a414>] handle_percpu_irq+0x50/0x80
[   22.428678] [<80055740>] generic_handle_irq+0x48/0x60
[   22.433847] [<803a153c>] do_IRQ+0x1c/0x2c
[   22.437953] [<80005e48>] except_vec_vi_end+0xc4/0xd0
[   22.443035] [<80005e48>] except_vec_vi_end+0xc4/0xd0
[   22.448107] handlers:
[   22.450474] [<81bc6fa0>] usb_hcd_irq [usbcore]
[   22.455015] Disabling IRQ #5
[   22.833800] irq 5: nobody cared (try booting with the "irqpoll" option)
```

**Comment by lantis1008** [comment]
Can i just check what you file you (compiled and?) flashed please?

**Comment by stopspazzing** [comment]
> Can i just check what you file you (compiled and?) flashed please?

```
Name: gargoyle_1.10.0-brcm47xx-generic-standard-squashfs.trx
Size: 4395008 bytes (4292 KiB)
CRC32: DDCBA6A3
CRC64: BE5F7BEF9D31C9E4
SHA256: 8D0BD93964427D0FB85258013B867CE8634CCEF11FED983392FCBC7FB85CE063
SHA1: AF489549A3B159844712ED24B4E53FBBE3EF0253
BLAKE2sp: 2C0C0FC787196E1CD38AA1063474D24AA0C2BE61D7CDC55E0FBF3633306C1DA1
```
```
Name: gargoyle_1.12.0-brcm47xx-generic-standard-squashfs.trx
Size: 6098944 bytes (5956 KiB)
CRC32: E23FD7F3
CRC64: D88E2D9C73F0B38C
SHA256: 31581643FE01655268C107C8A9AA6C5BBE2E91B91CB0A38E75522BFFC2BBBB79
SHA1: 3DAA6DF8131F541711DBB99D8E7D411298E2C908
BLAKE2sp: B9A8A79EF2460FFED30521C9EC3FFE53E3F504111AC7FA36B6AF25EA40A79B59
```
```
Name: gargoyle_1.13.x-brcm47xx-generic-standard-squashfs.trx
Size: 5902336 bytes (5764 KiB)
CRC32: F0D27BDC
CRC64: DBA7E5A22B9AD1CC
SHA256: 2C6E51701FC7DB582EAA73F0D782A2DD559D5381955079734E037A98A5246D3F
SHA1: 9DA27301C82561D95D73A8726DBD3595A25E30AB
BLAKE2sp: 53C6F520D57C454AC3BD068AFC462EAF9B4F336F113C6DAB7D8EBB0AA592D8E7
```

**Comment by lantis1008** [comment]
I don't know if it would make a difference, but e2000 is part of the mips24k subtarget, and you've flashed the generic version of the "standard" image.

**Comment by lantis1008** [comment]
Does this also not boot for you?
https://downloads.openwrt.org/releases/19.07.4/targets/brcm47xx/generic/openwrt-19.07.4-brcm47xx-generic-standard-squashfs.trx

Compared with this one
https://downloads.openwrt.org/releases/19.07.4/targets/brcm47xx/mips74k/openwrt-19.07.4-brcm47xx-mips74k-standard-squashfs.trx

**Comment by stopspazzing** [comment]
> Does this also not boot for you?
> https://downloads.openwrt.org/releases/19.07.4/targets/brcm47xx/generic/openwrt-19.07.4-brcm47xx-generic-standard-squashfs.trx
> 
> Compared with this one
> https://downloads.openwrt.org/releases/19.07.4/targets/brcm47xx/mips74k/openwrt-19.07.4-brcm47xx-mips74k-standard-squashfs.trx

Both work lol:
[generic-standard (first one) log](https://gist.github.com/stopspazzing/0a1605b76734b41cd51c955870fe16fc)

[mips74k-standard-squashfs (second one) log](https://gist.github.com/stopspazzing/9962c6a2c85197e51f10ab56e7f73de9)

**Comment by lantis1008** [comment]
At least that ties it down to being a Gargoyle specific issue. Thanks I'll have a look.

**Comment by lantis1008** [comment]
i've tried to move the config closer to the standard openwrt...
can you try this one please?
https://drive.google.com/open?id=1dKhnKsIxq3bUxKRlfJaFHgoDBy77xV77

**Comment by stopspazzing** [comment]
Yes, this one loads up just fine. There is no GUI but there isnt any errors or bootloops.

**Comment by lantis1008** [comment]
that's a bit unexpected. I didn't change anything that should have caused any significant fixes. 
Can you do a bit more poking on the lack of GUI please? All the necessary components are there. Is uHTTPd not running? Is the interface causing some error and failing to load?

**Comment by stopspazzing** [comment]
> that's a bit unexpected. I didn't change anything that should have caused any significant fixes.
> Can you do a bit more poking on the lack of GUI please? All the necessary components are there. Is uHTTPd not running? Is the interface causing some error and failing to load?


```
root@Gargoyle:/# ifconfig
lo        Link encap:Local Loopback
          inet addr:127.0.0.1  Mask:255.0.0.0
          inet6 addr: ::1/128 Scope:Host
          UP LOOPBACK RUNNING  MTU:65536  Metric:1
          RX packets:128 errors:0 dropped:0 overruns:0 frame:0
          TX packets:128 errors:0 dropped:0 overruns:0 carrier:0
          collisions:0 txqueuelen:1000
          RX bytes:7680 (7.5 KiB)  TX bytes:7680 (7.5 KiB)

```

only a loopback interface. Seems driver for network controller not loading?

**Comment by lantis1008** [comment]
Can you try another please?
https://drive.google.com/open?id=1dKhnKsIxq3bUxKRlfJaFHgoDBy77xV77

Think i left out the ethernet driver. I added a few more things back

**Comment by stopspazzing** [comment]
> Can you try another please?
> https://drive.google.com/open?id=1dKhnKsIxq3bUxKRlfJaFHgoDBy77xV77
> 
> Think i left out the ethernet driver. I added a few more things back

This fixed those issues. Fully working now and can access Gargoyle GUI. Everything seems to be running properly.

**Comment by lantis1008** [comment]
Thanks. If i do a 1.12 build do you mind testing that? I might as well try to fix that as well.
Based on a bit of digging around, it looks like including the USB drivers on some of the devices causes a problem and boot hangs.

**Comment by stopspazzing** [comment]
> Thanks. If i do a 1.12 build do you mind testing that? I might as well try to fix that as well.
> Based on a bit of digging around, it looks like including the USB drivers on some of the devices causes a problem and boot hangs.

Sure. lets get it done!

**Comment by stopspazzing** [comment]
> Thanks. If i do a 1.12 build do you mind testing that? I might as well try to fix that as well.
> Based on a bit of digging around, it looks like including the USB drivers on some of the devices causes a problem and boot hangs.

any update on this?

**Comment by lantis1008** [comment]
Been stuck updating qosmon for IPv6. I'll try to do it in the next few days

**Comment by lantis1008** [comment]
Sorry for the delay. Can you please try this one @stopspazzing ?
https://drive.google.com/file/d/1nmhVmjGpkTt15kLL0FienyC8gyFBQXbB/view?usp=sharing

**Comment by stopspazzing** [comment]
> Sorry for the delay. Can you please try this one @stopspazzing ?
> https://drive.google.com/file/d/1nmhVmjGpkTt15kLL0FienyC8gyFBQXbB/view?usp=sharing

Basic testing passed without issues. No out of ordinary errors/warnings in serial console on boot. Looks like you fixed the issue. Let me know when I can close this. 👍

**Comment by lantis1008** [comment]
Thanks for that. It is fixed as of f86c189.
I can't auto close the issue as it isn't resolved by a commit to master branch.
